cmd/tempo/app/app.go

```diff
@@ -58,154 +52,6 @@ var (
-	c.Compactor.RegisterFlagsAndApplyDefaults(util.PrefixConfig(prefix, "compactor"), f)
-	c.StorageConfig.RegisterFlagsAndApplyDefaults(util.PrefixConfig(prefix, "storage"), f)
-
-}
-
-// MultitenancyIsEnabled checks if multitenancy is enabled
-func (c *Config) MultitenancyIsEnabled() bool {
-	return c.MultitenancyEnabled || c.AuthEnabled
-}
-
-// CheckConfig checks if config values are suspect.
-func (c *Config) CheckConfig() {
-	if c.Target == MetricsGenerator && !c.MetricsGeneratorEnabled {
-		level.Warn(log.Logger).Log("msg", "target == metrics-generator but metrics_generator_enabled != true",
-			"explain", "The metrics-generator will only receive data if metrics_generator_enabled is set to true globally")
-	}
-
-	if c.Ingester.CompleteBlockTimeout < c.StorageConfig.Trace.BlocklistPoll {
-		level.Warn(log.Logger).Log("msg", "ingester.complete_block_timeout < storage.trace.blocklist_poll",
-			"explain", "You may receive 404s between the time the ingesters have flushed a trace and the querier is aware of the new block")
-	}
-
-	if c.Compactor.Compactor.BlockRetention < c.StorageConfig.Trace.BlocklistPoll {
-		level.Warn(log.Logger).Log("msg", "compactor.compaction.compacted_block_timeout < storage.trace.blocklist_poll",
-			"explain", "Queriers and Compactors may attempt to read a block that no longer exists")
-	}
-
-	if c.Compactor.Compactor.RetentionConcurrency == 0 {
-		level.Warn(log.Logger).Log("msg", "c.Compactor.Compactor.RetentionConcurrency must be greater than zero. Using default.", "default", tempodb.DefaultRetentionConcurrency)
-	}
-
-	if c.StorageConfig.Trace.Backend == "s3" && c.Compactor.Compactor.FlushSizeBytes < 5242880 {
-		level.Warn(log.Logger).Log("msg", "c.Compactor.Compactor.FlushSizeBytes < 5242880",
-			"explain", "Compaction flush size should be 5MB or higher for S3 backend")
-	}
-
-	if c.StorageConfig.Trace.BlocklistPollConcurrency == 0 {
-		level.Warn(log.Logger).Log("msg", "c.StorageConfig.Trace.BlocklistPollConcurrency must be greater than zero. Using default.", "default", tempodb.DefaultBlocklistPollConcurrency)
-	}
-
-	if c.Distributor.LogReceivedTraces {
-		level.Warn(log.Logger).Log("msg", "c.Distributor.LogReceivedTraces is deprecated. The new flag is c.Distributor.log_received_spans.enabled")
-	}
-}
-
-func (c *Config) Describe(ch chan<- *prometheus.Desc) {
-	ch <- metricConfigFeatDesc
-}
-
-func (c *Config) Collect(ch chan<- prometheus.Metric) {
-
-	features := map[string]int{
-		"search_external_endpoints": 0,
-		"search":                    0,
-		"metrics_generator":         0,
-	}
-
-	if len(c.Querier.Search.ExternalEndpoints) > 0 {
-		features["search_external_endpoints"] = 1
-	}
-
-	if c.SearchEnabled {
-		features["search"] = 1
-	}
-
-	if c.MetricsGeneratorEnabled {
-		features["metrics_generator"] = 1
-	}
-
-	for label, value := range features {
-		ch <- prometheus.MustNewConstMetric(metricConfigFeatDesc, prometheus.GaugeValue, float64(value), label)
-	}
-}
-
 func newDefaultConfig() *Config {
```

Comment:
I'd consider moving this function as well, this is kind of the constructor of `Config`.

Reply:
Oh good call.  I'll add that.